base/stream.jl

```diff
@@ -253,7 +253,7 @@ end
 
 function check_open(x::Union{LibuvStream, LibuvServer})
     if !isopen(x) || x.status == StatusClosing
-        throw(ArgumentError("stream is closed or unusable"))
+        throw(IOError("stream is closed or unusable", 0))
```

Comment:
It's a bit strange that the error code is 0, since that usually means "no error". Maybe we should pick something from the libuv error codes.

Reply:
Could use [`UV_ENOTCONN`](https://github.com/libuv/libuv/blob/81c77427afb39b6fa6c0ecffdc1c1e2b3dde7aea/include/uv.h#L120) or [`UV_EBADF`](https://github.com/libuv/libuv/blob/40498795ab457dfa3a00a5aad45ba9db3bf71588/src/unix/stream.c#L803-L805), but error handling code that checks for particular codes may rely on some particular semantic per the libuv docs. I chose `0` to mean "there is no uv error code for this `IOError`".

[`UV_UNKNOWN`](https://github.com/libuv/libuv/blob/e4087dedf837f415056a45a838f639a3d9dc3ced/include/uv.h#L138) would be another option.

My preference would be to remove the `code` field from `IOError` and then add subtypes (`AddirInUseError <: IOError`, `ConnResetError <: IOError`, etc) as needed for particular cases where error handling code needs to know the cause. e.g. #15879. (We could still [use the code to build the error string](https://github.com/JuliaLang/julia/blob/e4d933d6b1a9c47fbffc4651488154b517649f65/base/libuv.jl#L78), but not store it in the `IOError` struct.)

It doesn't seem quite right to expose the internal codes of the IO library we happen to be using today.